debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/launchConfigurations/LaunchConfigurationTabGroupViewer.java

```diff
@@ -842,8 +842,9 @@ protected void displayInstanceTabs(boolean redrawTabs) {
 			fInitializingTabs = false;
 			return;
 		}
+		Class<? extends ILaunchConfigurationTab> lastActiveTabKind = null;
```

Comment:
This should directly be initialized via `getActiveTab().getClass()` and not be retrieved from the `showInstanceTabsFor()` method. If I see correctly, that method must not handle that tab kind at all after removing the restore functionality from that method.

Reply:
That won't work: `disposeExistingTabs()` is called in between